### ISSUE TITLE
Unable to patch a resource

### ISSUE THREAD

**Issue by munjalpatel** [argoproj/argo-workflows#5310]
Hello,

I am trying to patch `sa` to add an `imagePullSecret`

Here is the template I am using:
```yaml
apiVersion: argoproj.io/v1alpha1
kind: ClusterWorkflowTemplate
metadata:
  name: add-image-pull-secret
spec:
  entrypoint: execute
  podGC:
    strategy: OnPodSuccess
  retryStrategy:
   retryPolicy: OnError
   limit: 1
  templates:
    - name: execute
      inputs: 
        parameters: [
          {name: "namespace"},
          {name: "secret", value: "github-container-registry"}
        ]
      resource:
        action: patch
        manifest: |
          apiVersion: v1
          kind: ServiceAccount
          metadata:
            name: default
            namespace: {{inputs.parameters.namespace}}
          imagePullSecrets:
          - name: {{inputs.parameters.secret}}
  tolerations:
    - effect: NoSchedule
      key: kubernetes.azure.com/scalesetpriority
      operator: Equal
      value: spot
```

When I run this, I keep getting this error:
```
executor error: Must provide at least one of flags or manifest.
```

Here is the full log from the step:
```
time="2021-03-05T14:41:14.433Z" level=info msg="Starting Workflow Executor" version="{latest+ae5587e.dirty 2021-03-02T21:48:50Z ae5587e97dad0e4806f7a230672b9
98fe140a767 v3.0.0-rc4 dirty go1.15.7 gc linux/amd64}"
time="2021-03-05T14:41:14.460Z" level=info msg="Creating a docker executor"
time="2021-03-05T14:41:14.460Z" level=info msg="Executor (version: latest+ae5587e.dirty, build_date: 2021-03-02T21:48:50Z) initialized (pod: ky-cloud-wor
kflows/provision-wp-st-mp-b6c56aa98d6a-1998521980) with template:\n{\"name\":\"execute\",\"inputs\":{\"parameters\":[{\"name\":\"namespace\",\"v
alue\":\"kcs-mp\"},{\"name\":\"secret\",\"value\":\"github-container-registry\"}]},\"outputs\":{},\"metadata\":{},\"resource\":{\"action\":\"patch\",\"me
rgeStrategy\":\"strategic\",\"manifest\":\"apiVersion: v1\\nkind: ServiceAccount\\nmetadata:\\n  name: default\\n  namespace: kcs-mp\\nimagePullSecrets:\
\n- name: github-container-registry\\n\"}}"
time="2021-03-05T14:41:14.460Z" level=info msg="Loading manifest to /tmp/manifest.yaml"
time="2021-03-05T14:41:14.461Z" level=error msg="executor error: Must provide at least one of flags or manifest.\ngithub.com/argoproj/argo-workflows/v3/error
s.New\n\t/go/src/github.com/argoproj/argo-workflows/errors/errors.go:49\ngithub.com/argoproj/argo-workflows/v3/workflow/executor.(*WorkflowExecutor).getKubec
tlArguments\n\t/go/src/github.com/argoproj/argo-workflows/workflow/executor/resource.go:101\ngithub.com/argoproj/argo-workflows/v3/workflow/executor.(*Workfl
owExecutor).ExecResource\n\t/go/src/github.com/argoproj/argo-workflows/workflow/executor/resource.go:31\ngithub.com/argoproj/argo-workflows/v3/cmd/argoexec/c
ommands.execResource\n\t/go/src/github.com/argoproj/argo-workflows/cmd/argoexec/commands/resource.go:48\ngithub.com/argoproj/argo-workflows/v3/cmd/argoexec/c
ommands.NewResourceCommand.func1\n\t/go/src/github.com/argoproj/argo-workflows/cmd/argoexec/commands/resource.go:25\ngithub.com/spf13/cobra.(*Command).execut
e\n\t/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:846\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\t/go/pkg/mod/github.com/spf13/cobra@v1.0.0/comman
d.go:950\ngithub.com/spf13/cobra.(*Command).Execute\n\t/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887\nmain.main\n\t/go/src/github.com/argoproj/arg
o-workflows/cmd/argoexec/main.go:14\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:204\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1374"
time="2021-03-05T14:41:14.461Z" level=fatal msg="Must provide at least one of flags or manifest.\ngithub.com/argoproj/argo-workflows/v3/errors.New\n\t/go/src
/github.com/argoproj/argo-workflows/errors/errors.go:49\ngithub.com/argoproj/argo-workflows/v3/workflow/executor.(*WorkflowExecutor).getKubectlArguments\n\t/
go/src/github.com/argoproj/argo-workflows/workflow/executor/resource.go:101\ngithub.com/argoproj/argo-workflows/v3/workflow/executor.(*WorkflowExecutor).Exec
Resource\n\t/go/src/github.com/argoproj/argo-workflows/workflow/executor/resource.go:31\ngithub.com/argoproj/argo-workflows/v3/cmd/argoexec/commands.execReso
urce\n\t/go/src/github.com/argoproj/argo-workflows/cmd/argoexec/commands/resource.go:48\ngithub.com/argoproj/argo-workflows/v3/cmd/argoexec/commands.NewResou
rceCommand.func1\n\t/go/src/github.com/argoproj/argo-workflows/cmd/argoexec/commands/resource.go:25\ngithub.com/spf13/cobra.(*Command).execute\n\t/go/pkg/mod
/github.com/spf13/cobra@v1.0.0/command.go:846\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\t/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950\ngithub
.com/spf13/cobra.(*Command).Execute\n\t/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887\nmain.main\n\t/go/src/github.com/argoproj/argo-workflows/cmd/
argoexec/main.go:14\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:204\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1374"
```

**Comment by terrytangyuan** [comment]
This seems to be a bug. I created https://github.com/argoproj/argo-workflows/pull/5311 to fix it. As a workaround, you can try providing some flags to your resource, e.g. `--record=false`.

**Comment by munjalpatel** [comment]
@terrytangyuan where/how can I provide these flags in the following manifest?

```yaml
apiVersion: argoproj.io/v1alpha1
kind: ClusterWorkflowTemplate
metadata:
  name: add-image-pull-secret
spec:
  entrypoint: execute
  podGC:
    strategy: OnPodSuccess
  retryStrategy:
   retryPolicy: OnError
   limit: 1
  templates:
    - name: execute
      inputs: 
        parameters: [
          {name: "namespace"},
          {name: "secret", value: "github-container-registry"}
        ]
      resource:
        action: patch
        manifest: |
          apiVersion: v1
          kind: ServiceAccount
          metadata:
            name: default
            namespace: {{inputs.parameters.namespace}}
          imagePullSecrets:
          - name: {{inputs.parameters.secret}}
  tolerations:
    - effect: NoSchedule
      key: kubernetes.azure.com/scalesetpriority
      operator: Equal
      value: spot
```

**Comment by terrytangyuan** [comment]
Flags can be supplied in `resource.flags` of your template. Here's an example: https://github.com/argoproj/argo-workflows/blob/master/examples/resource-flags.yaml

**Comment by munjalpatel** [comment]
@terrytangyuan when I add flag `--record=false` I now get the following error:
```
You must provide one or more resources by argument or filename.
```

Here is my updated template:
```yaml
apiVersion: argoproj.io/v1alpha1
kind: ClusterWorkflowTemplate
metadata:
  name: add-image-pull-secret
spec:
  entrypoint: execute
  podGC:
    strategy: OnPodSuccess
  retryStrategy:
   retryPolicy: OnError
   limit: 1
  templates:
    - name: execute
      inputs: 
        parameters: [
          {name: "namespace"},
          {name: "secret", value: "github-container-registry"}
        ]
      resource:
        action: patch
        flags: ["--record=false"]
        manifest: |
          apiVersion: v1
          kind: ServiceAccount
          metadata:
            name: default
            namespace: {{inputs.parameters.namespace}}
          imagePullSecrets:
          - name: {{inputs.parameters.secret}}
  tolerations:
    - effect: NoSchedule
      key: kubernetes.azure.com/scalesetpriority
      operator: Equal
      value: spot
```

**Comment by terrytangyuan** [comment]
Let's wait until https://github.com/argoproj/argo-workflows/pull/5311 gets merged and then you can try the latest images built from master branch.

**Comment by dinesh** [comment]
@alexec is this released in 3.0.2, I see the commit was not cherry-picked. though it's there in 2.12.11 release.

**Comment by wreed4** [comment]
This seems to still be broken, see details here: https://github.com/argoproj/argo-workflows/pull/5311#issuecomment-844541935

**Comment by tszngai** [comment]
Hi there in which release is this fixed? I'm using 3.0.2 and it's still broken. Thanks

**Comment by terrytangyuan** [comment]
The relevant fix should be https://github.com/argoproj/argo-workflows/commit/80f8473a13482387b9f54f9288f4a982a210cdea which is available in >3.1.0rc9
